### PR TITLE
Geth instead of ganache

### PR DESCRIPTION
**Pull request by musnit**
ganache crashes easily, not reliable especially with bigger more complex txs

this pr makes us use a local geth-based chain for testing directly

e2e tests pass :)!

### CODE REVIEW

File: test/genesis.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "eip150Block": 0,
+    "eip155Block": 0,
+    "eip158Block": 0,
+    "byzantiumBlock": 0,
+    "constantinopleBlock": 0,
+    "petersburgBlock": 0,
+    "istanbulBlock": 0,
+    "muirGlacierBlock": 0,
+    "berlinBlock": 0,
+    "londonBlock": 0,
```
[the remark this answers is not in the document]
Comment (vgeddes) [reply test/genesis.json:14]: Setting these to `0` seems to be the correct action as per https://github.com/ethereum/go-ethereum/blob/58b0420a8a5d7809981eab119ae64218ea53bc4c/params/config.go#L337

File: test/scripts/start-services.sh [hunk 2]
```diff
@@ -7,25 +7,17 @@ rm -rf $configdir
...
-        --chainId=344 \
-        --deterministic \
-        --db $configdir/ganache.db \
-        --mnemonic="stone speak what ritual switch pigeon weird dutch burst shaft nature shove" \
-        --gasLimit=8000000 \
-        >ganache.log 2>&1 &
-
-    scripts/wait-for-it.sh -t 32 localhost:8545
-    sleep 5
+kill $(ps -aux | grep -e geth -e hardhat -e polkadot/target -e ganache-cli -e release/snowbridge | awk '{print $2}') || true
```
Comment (vgeddes) [inline test/scripts/start-services.sh:10]: Lets remove `hardhat` and `ganache-cli` from the list.

File: test/genesis.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "04E00e6D2e9Ea1E2AF553De02A5172120BFA5c3e": { "balance": "100000000000000000000" },
+    "a255dC78C1510e2c1332fBAC2de848058f479CEE": { "balance": "100000000000000000000" },
+    "ACbd24742b87c34dED607FB87b22401B2Ede167E": { "balance": "100000000000000000000" },
+    "01F6749035e02205768f97e6f1d394Fb6769EC20": { "balance": "100000000000000000000" },
+    "8b66D5499F52D6F1857084A61743dFCB9a712859": { "balance": "100000000000000000000" },
+    "13e16C4e5787f878f98a610EB321170512b134D4": { "balance": "100000000000000000000" },
+    "eEBFA6B9242A19f91a0463291A937a20e3355681": { "balance": "100000000000000000000" },
+    "87D987206180B8f3807Dd90455606eEa85cdB87a": { "balance": "100000000000000000000" }
+  }
+}
```
Comment (vgeddes) [inline test/genesis.json:31]: Missing trailing newline.

**Review by vgeddes — APPROVED** [review]

File: ethereum/package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   "dependencies": {
     "@iarna/toml": "^2.2.5",
     "@openzeppelin/contracts": "^4.1.0",
-    "@openzeppelin/test-helpers": "^0.5.10",
+    "@openzeppelin/test-helpers": "https://github.com/Snowfork/openzeppelin-test-helpers.git",
```
Comment (philipstanislaus) [inline ethereum/package.json:9]: Depending on main/master may break tests in the future – can we lock this down to a specific commit at least?

File: test/key1.prv [hunk 5]
```diff
@@ -0,0 +1 @@
+d9c5dc47ed678fc3e63249953866d79e5cf48418e79d8eec1a985be7393ef3b9
```
Comment (philipstanislaus) [inline test/key1.prv:1]: Only cosmetic, but can we rename the files `test/key0.prv`, `test/key1.prv` and `test/password` to something looking less like exposed secrets? Might undermine trust in the project from someone superficially scanning the codebase. 

Maybe: 

`test/key0.prv` => `test/dev-example-key0.prv`
`test/key1.prv` => `test/dev-example-key1.prv`  
`test/password` =>  `test/dev-example-password`

**Review by philipstanislaus — COMMENTED** [review]
Nice, just a tiny nit :)

File: test/key1.prv (hunk 5 above)
Comment (musnit) [reply test/key1.prv:1]: https://github.com/Snowfork/snowbridge/pull/492

File: ethereum/package.json (hunk 4 above)
Comment (musnit) [reply ethereum/package.json:9]: https://github.com/Snowfork/snowbridge/pull/492

**Pull request merged**